### PR TITLE
send faux email address when creating registry credential 

### PR DESCRIPTION
**Pull request by wjimenez5271**
send faux email address when creating registry credential + logging enhancements

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package main
 import (
 	"encoding/base64"
 	"fmt"
-	"log"
+	log "github.com/Sirupsen/logrus"
```
Comment (cloudnautique) [inline main.go]: Typically just use `logrus` and no longer alias it to `log`.

Comment (wjimenez5271) [inline main.go]: yea my IDE freaks out about it without the alias and trys to import the built-in still...haven't figured out how to turn that off yet.

File: main.go [hunk 2]
```diff
@@ -30,6 +30,23 @@ type Rancher struct {
...
+	// check if config param has been set for log level, otherwise the default of the logrus package will be used
+	if log_level, ok := os.LookupEnv("LOG_LEVEL"); ok && log_level != "" {
+		log_level_obj, err := log.ParseLevel(log_level)
+		if err != nil {
+			log.Error(err)
+		}
+		log.SetLevel(log_level_obj)
+	}
+	// set log format to JSON
+	log.SetFormatter(&log.JSONFormatter{})
```
Comment (cloudnautique) [inline main.go]: what does this look like coming out of a container?

Comment (wjimenez5271) [inline main.go]: with latest update it now should look like
```
INFO[2017-03-14T15:08:01-07:00] A walrus appears                              animal=walrus
```

File: Makefile [hunk 3]
```diff
@@ -6,3 +6,6 @@ test:
 
 clean:
 	rm rancher-ecr-credentials
+
+push:
```
Comment (cloudnautique) [inline Makefile]: This is fine for now, but we should move this to Dapper and all the other stuff we do in Go-Skel.

File: main.go (hunk 1 above)
Comment (cloudnautique) [inline main.go]: You'll have to fix it if you work on any of our other projects :)

**Comment by cloudnautique** [comment]
OK, the logging change looks good, but there are a few things:

1. The tests fail I'm guessing because of the email change. 
2. main_test.go needs to be updated to use the mocks package from `rancher` not `objectpartners`
3. We should cut over to trash.yml for deps management
4. Convert this to dapper/makefile from go-skel.

All Rancher projects follow a similar pattern. You should be able to type make `deps,test,build,package,release,ci` etc. Releasing this based on someone elses flow is a bit painful.

Also, for future projects, please fork and merge in from your repo. We do not typically do branches in our main `rancher/*` projects unless its for release management.

**Comment by cloudnautique** [comment]
This needs the Dockerfile.dapper in the root directory from go-skel.

When its ready you should be able to:
make deps
make test
make build
make package

and it should all work.

**Pull request merged**